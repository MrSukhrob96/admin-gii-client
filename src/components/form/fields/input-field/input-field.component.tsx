import { FormGroup, InputGroup } from '@blueprintjs/core';
import { hoc } from '@utils/hoc';
import { Controller } from 'react-hook-form';
import { useInputFieldProps } from './input-field.props';

/**
 * <InputField />
 */
export const InputField = hoc(
  useInputFieldProps,
  ({ name, control, rules, helperText, label, ...rest }) => (
    <Controller
      name={name}
      control={control}
      rules={rules}
      render={({
        field: { value, onChange, onBlur, ref },
        fieldState: { error }
      }) => (
        <FormGroup
          helperText={error?.message || helperText}
          label={label}
          labelFor={`${name}-input`}
          labelInfo={rules?.required && '(majburiy)'}
          intent={error ? 'danger' : 'none'}
        >
          <InputGroup
            {...rest}
            name={name}
            value={value}
            onChange={onChange}
            onBlur={onBlur}
            id={`${name}-input`}
            inputRef={ref}
          />
        </FormGroup>
      )}
    />
  )
);
