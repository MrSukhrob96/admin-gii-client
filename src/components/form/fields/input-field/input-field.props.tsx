import { InputGroupProps2 } from "@blueprintjs/core";
import { Control, ControllerProps } from "react-hook-form";

/**
 * <InputField /> props
 */
export type InputFieldProps = {
  name: string;
  helperText?: string;
  label?: string;
  rules?: ControllerProps['rules'];
  control: Control<any>
} & InputGroupProps2

export const useInputFieldProps = (_?: InputFieldProps) => ({});

