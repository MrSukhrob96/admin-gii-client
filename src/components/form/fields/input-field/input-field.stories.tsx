import { ComponentStory, ComponentMeta } from '@storybook/react';
import { InputField } from './input-field.component';
import '@styles/index.css';

export default {
  title: 'Common/InputField',
  component: InputField,
} as ComponentMeta<typeof InputField>

const Template: ComponentStory<typeof InputField> = (args) => <InputField {...args} />;

export const InputFieldInitial = Template.bind({});
InputFieldInitial.args = {
}