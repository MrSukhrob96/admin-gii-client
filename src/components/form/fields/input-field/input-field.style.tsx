import styled from 'styled-components'

export const InputFieldContainer = styled.div`

`