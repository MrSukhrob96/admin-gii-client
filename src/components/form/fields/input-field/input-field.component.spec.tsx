import { customRender } from '@test-utils';
import { renderHook } from '@testing-library/react-hooks';
import { useForm } from 'react-hook-form';
import { InputField } from './input-field.component';

describe('InputField', () => {
  test('Should render InputField component', async () => {
    const { result } = renderHook(() => useForm({
      defaultValues: { test: '' }
    }));

    const { asFragment } = customRender(
      <InputField name='test' control={result.current.control} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});