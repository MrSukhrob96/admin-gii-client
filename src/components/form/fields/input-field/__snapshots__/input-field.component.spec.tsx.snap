// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField Should render InputField component 1`] = `
<DocumentFragment>
  <div
    class="bp4-form-group"
  >
    <div
      class="bp4-form-content"
    >
      <div
        class="bp4-input-group"
      >
        <input
          class="bp4-input"
          id="test-input"
          name="test"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
